src/System.Text.JsonLab/System/Text/Json/Utf8JsonWriter.WriteArrayValues.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.Text.JsonLab
+{
+    public ref partial struct Utf8JsonWriter2<TBufferWriter> where TBufferWriter : IBufferWriter<byte>
+    {
+        public void WriteArray(ReadOnlySpan<byte> propertyName, ReadOnlySpan<int> values)
+        {
+
+        }
+
+        public void WriteArray(ReadOnlySpan<byte> propertyName, ReadOnlySpan<long> values)
+        {
+            // TODO: Use throw helper with proper error messages
+            if (propertyName.Length > JsonConstants.MaxTokenSize || CurrentDepth >= JsonConstants.MaxDepth)
+                JsonThrowHelper.ThrowJsonWriterOrArgumentException(propertyName, _currentDepth);
+
+            if (_options.SlowPath)
+                WriteArraySlow(propertyName, values);
+            else
+                WriteArrayFast(propertyName, values);
+
+            _currentDepth |= 1 << 31;
+            _tokenType = JsonTokenType.EndArray;
+        }
+
+        private void WriteArrayFast(ReadOnlySpan<byte> propertyName, ReadOnlySpan<long> values)
+        {
+            if (values.Length > 0 && values.Length < (int.MaxValue - 2) / (JsonConstants.MaximumInt64Length + 1))
+            {
+                // Calculated based on the following: '[number0,number1,...,numberN]'
+                int bytesNeeded = 2 + values.Length * (1 + JsonConstants.MaximumInt64Length);
```

Comment:
Is there a concern with pessimistically asking `IBufferWriter<byte>` for more bytes than needed (i.e. get the bytes needed for the worst case)?

Reply:
cc @davidfowl, @pakrym - any thoughts on the approach of over-asking for the amount of bytes from IBufferWriter for the worst case? For instance, if we are writing a 4, single digit integers as a JSON array, I ask for 4*20 + 3 + 2 =85 bytes rather than 4 + 3+ 2 = 9 (for example: `[1, 2, 3, 4]`). This way I don't have to count the digits of each element (at most it would be 20 digits, for an Int64).

Reply:
Over-asking a little bit is fine and considering IBufferWriter is free to give you less than requested if it doesn't have enough space.